urllib3/connectionpool.py

```diff
@@ -381,8 +382,22 @@ def _make_request(self, conn, method, url, timeout=_Default,
         log.debug("\"%s %s %s\" %s %s" % (method, url, http_version,
                                           httplib_response.status,
                                           httplib_response.length))
+
+        header_parsing_errors = extract_parsing_errors(httplib_response.msg)
+        if header_parsing_errors:  # Platform-specific: Implementation.
+            logging.warning(
+                'Errors while parsing headers for %s %s: %s. Unparsed data: %s' %
```

Comment:
> I'd prefer to have the URL first, because the unparsed data may be big

Fair, sure.

Take a look at other logs for consistent formatting. As a rule of thumb, I usually write the explanation, then colon, then value. No quotes around the value (colon is easy enough to parse on if necessary).

Here's a similar example where we do url -> repr, here: `log.debug("Incremented Retry for (url='%s'): %r" % (url, new_retry))`  (IMO the "for" is extraneous here)

So how about: `"Failed to parse headers (url={0}): {1}".format(url, err)`